src/service.ts

```diff
@@ -373,14 +373,14 @@ export class Service {
     }
   }
 
-  static async deployContract(fiddleName: string, file: File, status?: IStatusProvider) {
+  static async deployContract(fiddleName: string, file: File, status?: IStatusProvider, contractSuffix: string = "") {
     gaEvent("deployContract", "Service", "wabt");
     const buffer = file.getData() as ArrayBuffer;
     try {
       status && status.push("Deploying contract");
       const config = await getConfig();
       return await this.postJson(`${config.contractHelper}/contract`, {
-        receiver: `studio-${fiddleName}`,
+        receiver: `studio-${fiddleName}${contractSuffix}`,
```

Comment:
I'd just pass `contractName` as parameter and then use `studio-${fiddleName}` for regular and `test-${timestamp}` for tests.

Reply:
I still like having fiddle id in this to make collisions less probable. studio-fiddlename-ttimestamp fits in char limit

Reply:
> I still like having fiddle id in this to make collisions less probable. studio-fiddlename-ttimestamp fits in char limit
 
it's ok, but let's still pass `contractName`